php/oop/medium/src/medium.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+class Post
+{
+    public function __construct(
+        private string $title,
+        private string $content,
+        private string $author,
+        private array $comments = []
+    ) {
+    }
+
+    public function addComment(string $string): void
+    {
+       $this->comments[] = $string;
+    }
+
+    public function showPostItem(string $requested_item)
+    {
+        return $this->$requested_item;
```

Comment:
Did you define `$this->requested_item` property in initialization?

Reply:
I don't use it like a property for Post class. It's just an argoman for showPostItem method.

Reply:
I didn't get it right. What does `showPostItem()` exactly?

Reply:
with showPostItem() method we can access to post data. we can't access to post's property because they are private and for accessing them i use this method.

Reply:
I didn't see the `$` 😄 
What if someone passes an invalid argument? Something like:
```php
$post = new Post('title', 'content', 'author');
$post->showPostItem('invalidProperty');
```
With the above example, the code will break. Actually, I suggest simple getter methods, dynamic and magics aren't good in general

Reply:
ohm 
it's true. I should work on it.